File: app/src/main/java/com/example/cb/test/ui/db/DbHelper.java
package com.example.cb.test.ui.db;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import androidx.annotation.Nullable;
import cb.xlibrary.utils.XLogUtils;

public class DbHelper extends SQLiteOpenHelper {

    public static String CREATE_STUDENT = "create table student("
            + "id integer primary key autoincrement,"
            + "name text,"
            + "age integer,"
            + "grade text)";


    public DbHelper(@Nullable Context context, @Nullable String name, @Nullable SQLiteDatabase.CursorFactory factory, int version) {
        super(context, name, factory, version);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(CREATE_STUDENT);
        XLogUtils.d("创建成果");
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

    }
}

File: app/src/main/java/com/example/cb/test/ui/db/DbHelper.kt
package com.example.cb.test.ui.db

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import cn.sccl.xlibrary.utils.XLogUtils

class DbHelper(context: Context?, name: String?, factory: SQLiteDatabase.CursorFactory?, version: Int) : SQLiteOpenHelper(context, name, factory, version) {

    companion object {
        var CREATE_STUDENT: String = ("create table student("
                + "id integer primary key autoincrement,"
                + "name text,"
                + "age integer,"
                + "grade text)")
    }

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(CREATE_STUDENT)
        XLogUtils.d("数据库创建成功")
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        XLogUtils.d("数据库更新-版本-->  $newVersion")
    }

}